### PR TITLE
consensus: verify payloads correctly

### LINKED ISSUE #555: consensus: refactor payload verification
Currently we just concatenate invocation and verification script and check results.
We should also do it as in 'Blockchain.verifyHashAgainstScript`.

### PR DESCRIPTION
**Pull request by fyrchik**
Verify script hash of the payload witness.

Closes #555 .

### CODE REVIEW

File: pkg/consensus/payload.go [hunk 1]
```diff
@@ -195,18 +196,34 @@ func (p *Payload) Sign(key *privateKey) error {
...
+	v.LoadScript(verification)
+	v.LoadScript(p.Witness.InvocationScript)
+
+	err := v.Run()
+	if err != nil || v.HasFailed() || v.Estack().Len() != 1 {
 		return false
 	}
 
-	result, err := v.Estack().Top().TryBool()
+	res, err := v.Estack().Pop().TryBool()
```
Comment (roman-khimov) [inline pkg/consensus/payload.go]: You can crash here with null pointer dereference.

File: pkg/consensus/payload.go [hunk 2]
```diff
@@ -195,18 +196,34 @@ func (p *Payload) Sign(key *privateKey) error {
...
+	if len(verification) == 0 {
+		bb := new(bytes.Buffer)
+		err := vm.EmitAppCall(bb, scriptHash, false)
+		if err != nil {
+			return false
+		}
+		verification = bb.Bytes()
+	} else if h := p.Witness.ScriptHash(); scriptHash != h {
+		return false
+	}
```
Comment (roman-khimov) [inline pkg/consensus/payload.go]: This chunk of code obviously reminds me of `verifyHashAgainstScript()`, I think we can make this code more reusable.

File: pkg/consensus/payload.go (hunk 1 above)
Comment (fyrchik) [reply pkg/consensus/payload.go]: I have checked if `Estack().Len()` is equal to one earlier.

Comment (roman-khimov) [reply pkg/consensus/payload.go]: OK, true.

**Pull request merged**